### ISSUE TITLE
markdown-layer hijacks standard text editing keybindings

### ISSUE THREAD

**Issue by aschmolck** [syl20bnr/spacemacs#9715]
`M-l dowcase-word`, `M-h mark-paragraph` and `M-k kill-sentence` are all standard text editing commands in (non-evil) emacs; rebinding them to demotion behavior by default seems like a pretty terrible idea.

http://spacemacs.org/layers/+lang/markdown/README.html#key-bindings

**Comment by Somelauw** [comment]
See also this issue https://github.com/syl20bnr/spacemacs/issues/9657

**Comment by humanfactors** [comment]
This has been open for quite some time and it's clearly in violation of the Spacemacs conventions. I'm finding it quite infuriating as I need to either maintain my own fork to address what is quite clearly a bug, or I'm required to go in and change the source each time (or even worse, hack on a solution in my dotfiles).

**Comment by humanfactors** [comment]
This is a pretty infuriating issue for me, particularly one that is so easy to address. I'm currently either having to maintain my own fork of just the markdown layer, or needing to manually adjust this in the source for every device I run Spacemacs on. Will this be fixed some time in the near future?

**Comment by syl20bnr** [comment]
Agree. But this is not so trivial to fix. Please give me the expected
functions under `M-h`, `M-j`, `M-k` and `M-l` and I'll make the
conventions smarter by settings these bindings only for Vim/Hybrid users
using hjkl navigation.

**Comment by humanfactors** [comment]
Thank you, Sylvain 👌! I'm currently using hybrid mode (so ideally would not over-ride the default Emacs bindings when in `evil-emacs-state-map`). I'm assuming the challenge is setting `markdown-mode-map` to be aware of the evilified state?

## Default Bindings
**`M-l`** ←  `M-l runs the command downcase-word (found in global-map), which is an interactive built-in function in ‘C source code’.`

**`M-k`** ← `M-k runs the command kill-sentence (found in global-map), which is an interactive compiled Lisp function in ‘paragraphs.el’.`

**`M-j`** ← `M-j runs the command indent-new-comment-line (found in global-map), which is an alias for ‘comment-indent-new-line’ in ‘newcomment.el’.`

**`M-h`**  ← `M-h runs the command mark-paragraph (found in global-map), which is an
interactive compiled Lisp function in ‘paragraphs.el’.`

**Comment by Somelauw** [comment]
Easy solution would be to integrate https://github.com/Somelauw/evil-markdown 
Originally I intended this package to become something like evil-org but for markdown. I haven't worked on it for a while, but it has worked well for me so far.

It's not on Melpa yet. I can make a PR to get it included or propose to add it to into https://github.com/emacs-evil/evil-collection.

In any case, I think it's better to have this issue solved by an external package instead of putting it in Spacemacs, so more people can use it.

**Comment by syl20bnr** [comment]
Should be fixed in develop branch, I also updated the conventions.

Thanks to @kccai for initial fix.

**Comment by raymestalez** [comment]
Hi everyone! I'm new to spacemacs. Does anybody have a fix for this very annoying problem?
Things like this make it extremely difficult and frustrating for a new user to switch to spacemacs.
There's a lot to learn, I really don't want to also struggle rewiring my muscle memory for the basic commonly used commands.
